packages/ag-charts-community/src/chart/interaction/highlightManager.ts

```diff
@@ -7,21 +7,28 @@ import type { SeriesNodeDatum } from '../series/seriesTypes';
  * distinct dependents and handles conflicting highlight requests.
  */
 export class HighlightManager {
-    private readonly highlightStates = new StateTracker<HighlightNodeDatum>();
+    private readonly highlightStates = new StateTracker<{
+        highlightedDatum?: HighlightNodeDatum;
+        previousHighlightedDatum?: HighlightNodeDatum;
+    }>();
```

Comment:
On reflection, this is the wrong pattern to use, `StateTracker` has a current state that is a combination of multiple inputs - conflating 'previous' state into the current state doesn't make logical sense.

Instead we should make use of the `highlight:change` event below to know when this has happened.